asteroid/dsp/spatial.py

```diff
@@ -2,64 +2,56 @@
-    The two signals can have different seq_lens but the reference signal should not be shorter than the input signal.
+def xcorr(inp, ref, normalized=True, eps=1e-8):
+    r"""Multi-channel cross correlation.
 
-    The length of output xcorr will be seq_len_ref - seq_len_input + 1.
+    The two signals can have different lengths but the input signal should be shorter than the reference signal.
 
-    Note: that the cross correlation is computed between each pair of microphone
-    channels and not between all possible pairs.
-    E.g. if input and ref have shape (2, 1, 100) the output will be (2, 1, 1) the first element is the xcorr between
-    the first mic channel of input and the first mic channel of ref.
-    If either input and ref have only one channel e.g. input: (3, 1, 100) and ref: (1, 1, 100) then output will be
-    (3, 1, 1) as ref will be broadcasted to have same shape as input.
+    .. note:: The cross correlation is computed between each pair of microphone channels and not
+        between all possible pairs e.g. if both input and ref have shape ``(1, 2, 100)``
+        the output will be ``(1, 2, 1)`` the first element is the xcorr between
+        the first mic channel of input and the first mic channel of ref.
+        If either input and ref have only one channel e.g. input: (1, 3, 100) and ref: ``(1, 1, 100)``
+        then output will be ``(1, 3, 1)`` as ref will be broadcasted to have same shape as input.
 
     Args:
-        input (:class:`torch.Tensor`): multi-channel input signal. Must be a tensor of shape (batch, mic_channels, seq_len).
-        ref (:class:`torch.Tensor`): multi-channel reference signal. Must be a tensor of shape (batch, mic_channels, seq_len)
+        inp (:class:`torch.Tensor`): multi-channel input signal. Shape: :math:`(batch, mic\_channels, seq\_len)`.
+        ref (:class:`torch.Tensor`): multi-channel reference signal. Shape: :math:`(batch, mic\_channels, seq\_len)`.
         normalized (bool, optional): whether to normalize the cross-correlation with the l2 norm of input signals.
         eps (float, optional): machine epsilon used for numerical stabilization when normalization is used.
 
     Returns:
         out (:class:`torch.Tensor`): cross correlation between the two multi-channel signals.
+            Shape: :math:`(batch, mic\_channels, seq\_len\_ref - seq\_len\_input + 1)`.
 
     """
-    # input has size batches, nmics2, seq_len2
-    # ref has size  batches, nmics1, seq_len1
-
-    assert input.size(0) == ref.size(0), "ref and input signals should have same batch size."
-    assert input.size(2) >= ref.size(2), "Input signal should not be shorter than the ref signal."
+    # inp: batch, nmics2, seq_len2 || ref: batch, nmics1, seq_len1
+    assert inp.size(0) == ref.size(0), "ref and inp signals should have same batch size."
+    assert inp.size(2) >= ref.size(2), "Input signal should be shorter than the ref signal."
```

Comment:
Do you agree that  "Input signal should not be shorter than the ref signal." was wrong?

Reply:
yes missing the not